pkg/daemon/ceph/osd/volume_test.go

```diff
@@ -260,6 +268,55 @@ var cephVolumeRAWTestResult = `{
 }
 `
 
+func TestInitializeBlockPVCResult(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
+		logger.Infof("%s %v", command, args)
+		if args[1] == "ceph-volume" && args[2] == "raw" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		} else if args[1] == "ceph-volume" && args[2] == "lvm" && args[3] == "prepare" && args[4] == "--bluestore" && args[7] == "--crush-device-class" {
+			return initializeBlockPVCTestResult, nil
+		}
+
+		return "", errors.Errorf("unknown command %s %s", command, args)
+	}
+	os.Setenv(oposd.CrushDeviceClassVarName, "foo")
+	defer os.Setenv(oposd.CrushDeviceClassVarName, "")
+	context := &clusterd.Context{Executor: executor}
+	clusterInfo := &cephconfig.ClusterInfo{
+		CephVersion: cephver.CephVersion{Major: 14, Minor: 2, Extra: 8},
+	}
+	a := &OsdAgent{cluster: clusterInfo, nodeName: "node1"}
+
+	devices := &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{
+			"data":     {Data: -1, Metadata: nil, Config: DesiredDevice{Name: "/mnt/set1-data-0-rpf2k"}},
+			"metadata": {Data: 0, Metadata: []int{1}, Config: DesiredDevice{Name: "/srv/set1-metadata-0-8c7kr"}},
+		},
+	}
+
+	blockPath, metadataBlockPath, err := a.initializeBlockPVC(context, devices, false)
+	assert.Nil(t, err)
+	assert.Equal(t, "/mnt/set1-data-0-rpf2k", blockPath)
+	assert.Equal(t, "/srv/set1-metadata-0-8c7kr", metadataBlockPath)
+
+	clusterInfo = &cephconfig.ClusterInfo{
+		CephVersion: cephver.Nautilus,
+	}
+	devices = &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{
+			"data":     {Data: -1, Metadata: nil, Config: DesiredDevice{Name: "/mnt/set1-data-0-rpf2k"}},
+			"metadata": {Data: 0, Metadata: []int{1}, Config: DesiredDevice{Name: "/srv/set1-metadata-0-8c7kr"}},
+		},
+	}
+	a = &OsdAgent{cluster: clusterInfo, nodeName: "node2"}
+	blockPath, metadataBlockPath, err = a.initializeBlockPVC(context, devices, false)
+	assert.Nil(t, err)
+	assert.Equal(t, "/dev/ceph-bceae560-85b1-4a87-9375-6335fb760c8c/osd-block-2ac8edb0-0d2e-4d8f-a6cc-4c972d56079c", blockPath)
+	assert.Equal(t, "/srv/set1-metadata-0-8c7kr", metadataBlockPath)
+
```

Comment:
Please also add a case in the unit tests when the OSD is already configured (that is, `device.Data != -1`) . It was there in one of the previous commits. But missing in the latest commit.

Reply:
yes, actually in that we have to assert empty that's why I left but yes I updated this in the next PR.